packages/layers/base/app/services/business/index.ts

```diff
@@ -0,0 +1,28 @@
+export * from './mutations'
+export * from './queries'
+export * from './query-keys'
+
+export const useBusinessService = () => {
+  const query = useBusinessQuery()
+
+  /**
+   * Fetches the business info of the current business.
+   * @param businessId the identifier of the business
+   * @param slim return BusinessDataSlim or BusinessData
+   * @returns a promise to return business data
+   */
+  function getBusiness(businessId: string, slim: true): Promise<BusinessDataSlim>
+  function getBusiness(businessId: string, slim?: false): Promise<BusinessData>
```

Comment:
Thats a neat way of defining dynamic response types. As long as folks understand overloading it is probably more intuitive that union response types